src/ClassicUO.Client/Game/UI/Gumps/RaceChangeGump.cs

```diff
@@ -410,39 +415,68 @@ private void ColorPicker_ColorSelected(object sender, ColorSelectedEventArgs e)
 
             CurrentColorOption[e.Layer] = new Tuple<int, ushort>(e.SelectedIndex, e.SelectedHue);
 
-            if (e.Layer != Layer.Invalid)
-            {
-                Item item = fakeMobile.FindItemByLayer(e.Layer);
-
-                if (item != null)
-                {
-                    item.Hue = e.SelectedHue;
-                }
-            }
-            else
+            switch (e.Layer)
             {
-                fakeMobile.Hue = e.SelectedHue;
+                case Layer.Beard:
+                    beard.Hue = e.SelectedHue;
+                    break;
+                case Layer.Hair:
+                    hair.Hue = e.SelectedHue;
+                    break;
+                case Layer.Invalid:
+                    fakeMobile.Hue = e.SelectedHue;
+                    break;
             }
 
-            paperDollInteractable.RequestUpdate(hair, beard);
+            paperDollInteractable.RequestUpdate();
         }
 
         private void ConfirmButton_MouseUp(object sender, MouseEventArgs e)
         {
-            if (e.Button == Input.MouseButtonType.Left)
-            {
-                NetClient.Socket.Send_ChangeRaceRequest(
-                    CurrentColorOption[Layer.Invalid].Item2,
-                    (ushort)CharacterCreationValues.GetHairComboContent(isFemale, selectedRace).GetGraphic(CurrentOption[Layer.Hair]),
-                    CurrentColorOption[Layer.Hair].Item2,
-                    (ushort)CharacterCreationValues.GetFacialHairComboContent(selectedRace).GetGraphic(CurrentOption[Layer.Beard]),
-                    CurrentColorOption[Layer.Beard].Item2
-                );
+            foreach (var co in CurrentColorOption)
+            {
+                Console.WriteLine($"{co.Key}: {co.Value}");
+            }
```

Comment:
To remove I guess

Reply:
You are correct, good catch. Also removed the unnecessary try/catch